tiledb/sm/rest/rest_client_remote.cc

```diff
@@ -1516,4 +1516,23 @@ RestClientRemote::post_consolidation_plan_from_rest(
       serialization_type_, returned_data);
 }
 
+const RestCapabilities& RestClientRemote::get_capabilities_from_rest() {
+  // Return early if already detected REST capabilities for this REST client.
+  if (rest_capabilities_.detected_) {
+    return rest_capabilities_;
+  }
+
+  // Init curl and form the URL
+  Curl curlc(logger_);
+  throw_if_not_ok(curlc.init(
+      config_, extra_headers_, &redirect_meta_, &redirect_mtx_, false));
+  const std::string url = rest_server_ + "/v4/capabilities";
+
+  Buffer data;
+  throw_if_not_ok(curlc.get_data(stats_, url, serialization_type_, &data, {}));
+  rest_capabilities_ =
+      serialization::rest_version_deserialize(serialization_type_, data);
+  return rest_capabilities_;
```

Comment:
```suggestion
return serialization::rest_version_deserialize(serialization_type_, data);
```

Reply:
We need to set `rest_capabilities_` before returning or it will never be initialized and we'll make the request every time we access it.

Reply:
Oops sorry, I missed that.